### ISSUE TITLE
Checkout AddressType differentBillingAddress logic is not working as expected

### ISSUE THREAD

**Issue by egils** [Sylius/Sylius#11617]
**Sylius version affected**: 1.7.5

**Description**  
Shipping and billing addresses are set to empty when using shipping address as default one (and different billing address) in checkout address step. This bug was introduced with https://github.com/Sylius/Sylius/pull/11033. It is caused by using ```isset($orderData['billingAddress'])``` in [AddressType.php](https://github.com/Sylius/Sylius/blame/master/src/Sylius/Bundle/CoreBundle/Form/Type/Checkout/AddressType.php#L126). Both form elements for shipping and billing addresses are submitted in checkout even tho one part of a form is hidden from user and all fields are submitted as empty text fields. Both array keys (```billingAddress`` and ```shippingAddress```) are always set when using this form type in twig therefore ```isset``` check is not sufficient.

**Steps to reproduce**  
Change checkout form template so shippingAddress is rendered first and billing is hidden, render ```form_row(form.differentBillingAddress)``` instead of ```form_row(form.differentShippingAddress)```, go to checkout address step, fill the shipping form, do NOT check the different address checkbox, submit form.

**Possible Solution**  
Change the twig/js logic in a way that hidden part of a form is not submitted.

**Comment by lchrusciel** [comment]
Hey @egils,

thanks for raising an issue. According to your proposed solution, Symfony by default will null all fields that are declared on the form type but are not rendered. I'm not sure how can we affect this behavior. Do you have some idea about how to achieve what you suggested? Are you willing to contribute this fix?

Also, recently we merged #11367. Can you check if the problem still exists with the recent changes?

**Comment by vvasiloi** [comment]
@lchrusciel that PR won't solve this issue. Someone mentioned it in Slack, about a month ago. I will try to find it.

**Comment by vvasiloi** [comment]
Found it: https://sylius-devs.slack.com/archives/C3EGDG9LY/p1590675709443000
The link will probably become inactive soon, so here's a
<details>
  <summary>screenshot</summary>
   
![sylius_address_issue](https://user-images.githubusercontent.com/7114562/86228941-b2a78400-bb97-11ea-96c6-155f763de3f1.png)

</details>

**Comment by egils** [comment]
@vvasiloi yes, this is exactly my problem as I changed default Sylius checkout form to use shipping address as primary and expected billing address data to be copied from shipping address if `differentBillingAddress` checkbox is not checked. 

@lchrusciel I have no idea how to easily fix this issue. What I did as a "temporary" hack for now on my project is setting the second form field to disabled by default and use JS to re-enable/disable them on checkbox state change. In this case the disabled fields data is not submitted and `isset()` check is valid. However, I do not consider it to be a valid long-term solution.

**Comment by demorose** [comment]
Hi. Emmanuel L here.
I fixed it in my project by having this js:
```js
const addressForm = document.querySelector('form[name="sylius_checkout_address"]');

if(addressForm) {
    const billingAddressPrefix = 'sylius_checkout_address[billingAddress]';
    const shippingAddressPrefix = 'sylius_checkout_address[shippingAddress]';
    const fieldKeys = ['firstName', 'lastName', 'phoneNumber', 'company', 'street', 'city', 'postcode', 'countryCode', 'latitude', 'longitude'];

    addressForm.addEventListener('submit', (e) =>{
        const formData = new FormData(addressForm);
        // Does user specifically want to override the address
        const changeBillingToShipping = addressForm.querySelector('*[name="sylius_checkout_address[differentBillingAddress]"]:not(:checked)');
        const changeShippingToBilling = addressForm.querySelector('*[name="sylius_checkout_address[differentShippingAddress]"]:not(:checked)');

        if ( changeBillingToShipping ) {
            fieldKeys.forEach( input => {
                console.log('change billing ', input);
                addressForm.querySelector('*[name="' + billingAddressPrefix + '[' + input + ']"]').value = formData.get(shippingAddressPrefix + '[' + input + ']');
            })
        }

        if ( changeShippingToBilling ) {
            fieldKeys.forEach( input => {
                console.log('change shipping ', input);
                addressForm.querySelector('*[name="' + shippingAddressPrefix + '[' + input + ']"]').value = formData.get(billingAddressPrefix + '[' + input + ']');
            })
        }
    });
}
```

In php, we should differenciate if the user did not check the checkbox or if the checkbox was not an option for him.
Maybe a hidden field with value "false" and the checkbox with value "true" that overwrite the hidden field value?

**Comment by jbcr** [comment]
This PR https://github.com/Sylius/Sylius/pull/12240 solve the problem.

`array_filter` remove in address all empty field. If the result array is empty, the address is empty.

**Comment by maximehuran** [comment]
It cannot work because some data can be sent (select with required value for example).

```
"billingAddress" => [▼
    "company" => ""
    "gender" => "u"
    "firstName" => ""
    "lastName" => ""
    "street" => ""
    "countryCode" => "FR"
    "provinceName" => ""
    "postcode" => ""
    "city" => ""
    "phoneNumber" => ""
  ]
```

**Comment by NoResponseMate** [comment]
It seems it has been inadvertently fixed in 1.12 during the [checkout required address](https://github.com/Sylius/Sylius/pull/14200) rework.

Cheers :beers:
